va/tlsalpn.go

```diff
@@ -287,6 +288,10 @@ func (va *ValidationAuthorityImpl) validateTLSALPN01(ctx context.Context, identi
 					hex.EncodeToString(h[:]),
 				))
 			}
+			// We were successful, so record the negotiated key exchange mechanism in
+			// the last validationRecord.
+			// TODO(#7321): Remove this when we have collected enough data.
+			validationRecords[len(validationRecords)-1].UsedRSAKEX = usedRSAKEX(cs.CipherSuite)
```

Comment:
There's only one validationRecord for TLS-ALPN-01 because there are no redirects. Earlier in this function we make the assumption there's exactly one by subscripting `[0]`. We should document that assumption earlier in this function, assign to a singular `validationRecord`, and use it here (and remove the verbiage about "last").

Reply:
Heh I actually had it that way at first, and then changed it to be a copy-paste of the http-01 code for the sake of consistency. Your suggestion of keeping a local reference to the singular validationRecord is actually what `tryGetChallengeCert` already does -- but that function only has a single caller, so I've simplified it to only return a single validationRecord, and used its strategy in this function instead.